src/psyclone/dynamo0p3.py

```diff
@@ -8667,6 +8617,76 @@ def ref_name(self, function_space=None):
             "DynKernelArgument.ref_name(fs): Found unsupported argument "
             "type '{0}'.".format(self._argument_type))
 
+    def _init_data_type_properties(self):
```

Comment:
I would expect a test in dynamo0p3_test.py for this function which checks that the values are initialised as expected e.g. precision, datatype, proxy_datatype have expected values. It could be part of an initialisation test for DynKernelArgument. However, given that there is little of this approach in dynamo0p3 and you have 100% coverage I'm willing to ignore this for this PR.

Reply:
OK, I take it back. You have done this, just in different files. So ignore me. When we finally restructure this par of the code I think it will be nicer to have a one-to-one correspondence between files and test files, but that is not your problem :-)

Reply:
Of course I have 100% coverage 😄 Re tests in different files, we moved things from `dynamo0p3_test.py` because it was growing and it is still quite large.